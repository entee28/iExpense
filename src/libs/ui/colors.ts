export default {
  primary5: '#F7FAFF',
  primary10: '#E5EEFF',
  primary20: '#0055FF33',
  primary30: '#99BBFF',
  primary40: 'rgba(0, 85, 255, 0.4)',
  primary50: 'rgba(0, 85, 255, 0.5)',
  primary80: '#3377FF',
  primary100: '#0055FF',
  mono5: '#1A1A1A0D',
  mono4: '#1A1A1A0A',
  mono10: '#1A1A1A1A',
  mono40: '#1A1A1A66',
  mono60: '#1A1A1A99',
  mono70: '#1A1A1AB2',
  mono80: '#1A1A1ACC',
  mono100: '#1A1A1A',
  warning5: '#FFA8000D',
  warning10: '#FFA8001A',
  warning30: '#FFA8004D',
  warning80: '#FFA80080',
  warning100: '#FFA800',
  warning: '#FF6100',
  destructive: '#E60A32',
  destructive80: '#E60A32CC',
  destructive30: '#E60A324D',
  destructive30Hard: '#F8B6C1',
  destructive10: '#E60A321A',
  destructive5: 'rgba(230, 10, 50, 0.05)',
  positive5: '#00B47E0D',
  positive10: '#00B47E1A',
  positive30: '#00B47E4D',
  positive100: '#00B47E',
  white: '#FFFFFF',
  transparent: 'transparent',
  aliceBlue: '#EBF3FF'
}
