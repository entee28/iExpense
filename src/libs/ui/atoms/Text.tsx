import React, { useMemo } from 'react'
import { TextProps, TextStyle, Text as RNText } from 'react-native'
import colors from '../colors'

type Props = {
  children?: React.ReactNode
  textProps?: TextProps
  extraBold?: boolean
  bold?: boolean
  semiBold?: boolean
  medium?: boolean
  regular?: boolean
} & TextStyle

export enum FontFamily {
  extraBold = 'Nunito-ExtraBold',
  bold = 'Nunito-Bold',
  medium = 'Nunito-Medium',
  semiBold = 'Nunito-SemiBold',
  regular = 'Nunito-Regular'
}

export const Text = ({
  textProps,
  children,
  fontSize = 14,
  lineHeight = 22,
  color = colors.mono100,
  textAlign = 'auto',
  extraBold,
  bold,
  semiBold,
  medium,
  regular,
  ...styles
}: Props) => {
  const fontFamily = useMemo(() => {
    if (extraBold) {
      return FontFamily.extraBold
    }
    if (bold) {
      return FontFamily.bold
    }
    if (semiBold) {
      return FontFamily.semiBold
    }
    if (medium) {
      return FontFamily.medium
    }
    if (regular) {
      return FontFamily.regular
    }
    return FontFamily.regular
  }, [extraBold, bold, semiBold, medium, regular])

  return (
    <RNText
      {...textProps}
      style={[{ color, textAlign, lineHeight, fontSize, fontFamily }, styles]}>
      {children}
    </RNText>
  )
}
