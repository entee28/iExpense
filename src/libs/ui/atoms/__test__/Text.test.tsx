import { render } from '@testing-library/react-native'
import colors from 'libs/ui/colors'
import { Text } from '../Text'

it('renders text correctly', () => {
  const text = render(<Text>Hello World</Text>)

  expect(text.container.props.children).toEqual('Hello World')
})

it('renders the correct styles', () => {
  const text1 = render(<Text>Hello World</Text>)
  const text2 = render(
    <Text
      color={colors.mono40}
      fontSize={16}
      lineHeight={24}
      textAlign="center"
      bold>
      Hello World
    </Text>
  )

  const text3 = render(<Text semiBold>Hello World</Text>)

  expect(text1.getByText('Hello World')).toHaveStyle({
    color: colors.mono100,
    fontSize: 14,
    lineHeight: 22,
    textAlign: 'auto',
    fontFamily: 'Nunito-Regular'
  })

  expect(text2.getByText('Hello World')).toHaveStyle({
    color: colors.mono40,
    fontSize: 16,
    lineHeight: 24,
    textAlign: 'center',
    fontFamily: 'Nunito-Bold'
  })

  expect(text3.getByText('Hello World')).toHaveStyle({
    fontFamily: 'Nunito-SemiBold'
  })
})

it('renders correctly', () => {
  const snapshot = render(
    <Text color="red" fontSize={14} lineHeight={16} medium>
      Hello World
    </Text>
  ).toJSON()

  expect(snapshot).toMatchSnapshot()
})
