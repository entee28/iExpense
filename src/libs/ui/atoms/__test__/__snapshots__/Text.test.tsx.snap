// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "red",
        "fontFamily": "Nunito-Medium",
        "fontSize": 14,
        "lineHeight": 16,
        "textAlign": "auto",
      },
      {},
    ]
  }
>
  Hello World
</Text>
`;
